### DISCUSSION TITLE
Is it possible to create multiple rules in one yaml file

### DISCUSSION THREAD

**Discussion by amitgupta369** [StackStorm/st2#5808] [Q&A] [answered]
Hi, 

Is it possible to create multiple rules in one yaml file with separated by --- .

If i am trying to create two rules in one yaml file then i am getting Failed to registered rule ... expected a single document in the stream in .. but found another document.

Thanks in advance.

**Comment by ip-sf** [comment, marked as answer]
You cannot. A yaml file is a single structured object document. `---` denotes the content of the file implicitly and does not act as a break or similar sequence. Stackstorm depends on well known keys to load data, and supplying a second rule in the same document would cause a duplicate key error, were it to even parse, which I assume is what is causing your error.

1 rule 1 file.
